### ISSUE TITLE
experimental_createPersister cache not cleared from Dev Tools

### ISSUE THREAD

**Issue by mozeryansky** [TanStack/query#6269]
### Describe the bug

When I pressed the clear cache icon in the dev tools I see the queries disappear but the cache is still present, so the next time I run the queries they're already cached.

### Your minimal, reproducible example

ttps://codesandbox.io/

### Steps to reproduce

```
import { experimental_createPersister } from '@tanstack/react-query-persist-client'
import { del, get, set } from 'idb-keyval'

export function createIDBPersister() {
  return experimental_createPersister({
    storage: {
      getItem: get,
      setItem: set,
      removeItem: del,
    },
    maxAge: 24 * 60 * 60 * 1000,
  })
}
```

### Expected behavior

Cache is cleared by calling the removeItem function

### How often does this bug happen?

None

### Platform

Firefox

### Tanstack Query adapter

react-query

### TanStack Query version

5.0.5

**Comment by TkDodo** [comment]
> the queries disappear but the cache is still present

what don you mean with that? Clearing the cache only clears the on memory cache, not the persisted storage. That is on purpose because it allows you to store data for longer on the storage than you have it in memory. 

If you want your storage to be a 1:1 representation of your in-memory cache, use `PersistQueryClientProvider`

**Comment by mozeryansky** [comment]
That makes sense. I was assuming the trash icon removed all cache. I was using clear for testing, so when I added in a persister I didn't realize it would be different.

I had switched from using `PersistQueryClientProvider` to a persister because I wanted to have some queries use in-memory and others persist to the Indexed DB. Also I have a lot of data, and the persistor does key/value pairs by query instead of the entire client like the provider does. I'm not sure of the performance difference through.
